File: CS/Modules.Projects/ProjectModule.cs
using Prism.Modularity;
using Prism.Regions;

namespace Modules.Projects {
    public class ProjectModule : IModule {
        private readonly IRegionManager regionManager;
        public ProjectModule(IRegionManager regionManager) {
            this.regionManager = regionManager;
        }
        void IModule.Initialize() {
            regionManager.RegisterViewWithRegion("ProjectsRegion", typeof(ProjectView));
        }
    }
}

File: VB/Modules.Projects/ProjectModule.vb
Imports Prism.Regions
Imports Prism.Modularity

Namespace Modules.Projects
    Public Class ProjectModule
        Implements IModule
        Private ReadOnly regionManager As IRegionManager
        Public Sub New(ByVal regionManager As IRegionManager)
            Me.regionManager = regionManager
        End Sub
        Public Sub Initialize() Implements IModule.Initialize
            regionManager.RegisterViewWithRegion("ProjectsRegion", GetType(ProjectView))
        End Sub
    End Class
End Namespace
